src/interfaces/wallet.cpp

```diff
@@ -330,14 +316,13 @@ class WalletImpl : public Wallet
         if (mi == m_wallet->mapWallet.end()) {
             return false;
         }
-        if (Optional<int> height = locked_chain->getHeight()) {
-            num_blocks = *height;
-            block_time = locked_chain->getBlockTime(*height);
+        num_blocks = m_wallet->GetLastBlockHeight();
+        if (num_blocks >= 0) {
+            block_time = m_wallet->chain().getBlockTime(num_blocks);
```

Comment:
I think to avoid the race condition mentioned https://github.com/bitcoin/bitcoin/pull/16426#discussion_r357761094, where there is a block reorg between the `CWallet::GetLastBlockHeight` and `Chain::getBlockTime` calls, it would be safer to avoid triggering asserts by getting the time by hash rather than height. You could tweak `GetLastBlock` to optionally return the block hash:

```c++
int GetLastBlock(uint256* block_hash = nullptr);
```

for

```c++
uint256 block_hash;
num_blocks = m_wallet->GetLastBlock(&block_hash);
if (num_blocks < 0 || !m_wallet.chain().findBlock(block_hash, nullptr, &block_time)) {
    block_time = -1;
}
```

There are probably other places in this PR where similar changes would be appropriate.

Reply:
Changed it in cbd2c1a, covered with other reorg mitgations.